cfe_internal/update/update_policy.cf

```diff
@@ -597,4 +610,64 @@ body delete u_tidy
       rmdirs   => "true";
 }
 
+bundle agent install_modules
+{
+  vars:
+    "vendored_dir" string => "$(sys.inputdir)$(const.dirsep)modules$(const.dirsep)packages$(const.dirsep)vendored$(const.dirsep)";
+    "override_dir" string => "$(sys.inputdir)$(const.dirsep)modules$(const.dirsep)packages$(const.dirsep)";
+    "vendored_paths" slist => findfiles("$(vendored_dir)*.mustache");
+    "package_paths_tmp" slist => findfiles("$(override_dir)*");
+    "package_paths" slist => filter("$(override_dir)vendored", package_paths_tmp, "false", "true", 999);
+    "vendored_modules" slist => maplist(regex_replace("$(this)", "$(vendored_dir)(.*).mustache", "$1", "g"), @(vendored_paths));
+    "override_modules" slist => maplist(regex_replace("$(this)", "$(override_dir)(.*)", "$1", "g"), @(package_paths));
+
+  classes:
+    "override_vendored_module_$(override_modules)" expression => fileexists("$(override_dir)$(override_modules)");
+    "module_$(override_modules)" expression => fileexists("$(override_dir)$(override_modules)");
+
+# NOTE: here we only send `bindir` into the template
+# this helps since we also rely on makefiles to replace
+# the python shebang for tests, so need to limit "what we change"
+# in these packages modules, instead of providing "everything"
+# we could send to the template via `datastate()`
+
+  files:
+    "$(sys.workdir)/modules/packages/$(vendored_modules)"
+      create => "true",
+      perms => u_mo("755", "root"),
```

Comment:
Oh, my spidy sense went off about SELINUX.

Reply:
How so?

Reply:
I need to check and make sure the perms are even needed. Maybe handled better elsewhere and if so let's keep it DRY. :)

Reply:
Right. This is handled in that other promise which you wondered if we should take out. ;)

https://github.com/cfengine/masterfiles/blob/master/cfe_internal/update/update_policy.cf#L248